roles/download/tasks/nc_apps.yml

```diff
@@ -0,0 +1,38 @@
+---
+- name: nc_apps | Parse the item values
+  ansible.builtin.set_fact:
+    nc_app_name: "{{ item.value if item.value != None else item.key }}"
```

Comment:
hello there.
you don't need to use set_facts for one time use variables. You can just declare it as ad-hoc var in the task below.
set_fact have a high level of precedence in the var hierarchy of ansible. furthermore, having the "fact" being changed at every iteration of the loop could create issues.
```suggestion
# add this in the task below instead
  vars:
    nc_app_name: "{{ item.value if item.value != None else item.key }}"
```